File: Domain/ResponseModels/BaseOperationState.cs
using Domain.BusinessEnums;

namespace Domain.ResponseModels
{
    public class BaseOperationState
    {
        public int IdOperation { get; set; }
        public string VideoName { get; set; }
        public string OperationType { get; set; }
        public string OperationStatus { get; set; }
        public int ProgressPercentage { get; set; }
    }
}

File: MovieCutterWebApp/src/app/shared/models/api-models/operation/base-operation-state.ts
import { OperationStatus } from '@shared/enums/operation-status';
import { OperationType } from '@shared/enums/operation-type';

export interface BaseOperationState {
  idOperation: number;
  videoName: string;
  operationType: OperationType;
  operationStatus: OperationStatus;
  progressPercentage: number;
}
